src/util/pelz_json_parser.c

```diff
@@ -37,15 +37,15 @@ int request_decoder(charbuf request, RequestType * request_type, charbuf * key_i
   switch (*request_type)
   {
   case REQ_ENC:
-    if (encrypt_parser(json, key_id, data))
+    if (encrypt_parser(json, key_id, data, request_sig, requestor_cert))
```

Comment:
If we add a new request type for a signed wrap request (e.g., REQ_ENC_SIGNED), then we could put the 'case' for that request type above it. Within that 'case', we could validate the requestor's certificate and verify the signature over the request. On failure of either of these we could error (e.g., log the error and return an appropriate error response message to the requestor). If both checks are successful, however, if we omit a 'break', we can just fall through and process as an unsigned wrap request (REQ_ENC).